server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleExtensionDoc.java

```diff
@@ -39,8 +39,7 @@ public RuleExtensionDoc() {
 
   @Override
   public String getId() {
-    // the Id is generated by elasticsearch
-    return null;
+    return getRuleKey() + "|" + getScope().getScope();
```

Comment:
IntelliJ inspections tell me, that this code is not faster as the string concatenation:

```
    return new StringBuilder(getRuleKey().toString())
      .append('|')
      .append(getScope().getScope())
      .toString();

```

So I'll leave everything as it is. Any objections?